### ISSUE TITLE
cl.ParallelogramOLF to support cumulative factors

### ISSUE THREAD

**Issue by kennethshsu** [casact/chainladder-python#922]
Consider this example in column 11:

<img width="1023" height="366" alt="Image" src="https://github.com/user-attachments/assets/79b6f7ff-379d-428c-8073-694e955d939c" />

We can use the cl.ParallelogramOLF to trend, with a bit of gymnastic.

> ```python
> friedland_gl_self_insurer = cl.load_sample("friedland_gl_self_insurer")
> 
> tort = pd.DataFrame({
>     "EffDate": ["2003-01-01", "2004-01-01", "2005-01-01"],
>     "RateChange": [(0.67/0.75) - 1, 0.75-1, 0.00]
> })
> 
> olf = cl.ParallelogramOLF(
>     rate_history=tort,
>     change_col="RateChange",
>     date_col="EffDate",
>     policy_length=12,
>     approximation_grain="M",
>     vertical_line=True
> ).fit_transform(friedland_gl_self_insurer["Reported Claims"]).olf_
> olf
> ```
> ```python
> 	2008
> 1998	0.6700
> 1999	0.6700
> 2000	0.6700
> 2001	0.6700
> 2002	0.6700
> 2003	0.7500
> 2004	1.0000
> 2005	1.0000
> 2006	1.0000
> 2007	1.0000
> 2008	1.0000
> ``` 

Can we make the rate change a bit easier?

 _Originally posted by @kennethshsu in [#582](https://github.com/casact/chainladder-python/issues/582#issuecomment-4626880017)_

**Comment by henrydingliu** [comment]
can we see a test?

**Comment by kennethshsu** [comment]
huh? What test?

**Comment by henrydingliu** [comment]
a test for this new feature

**Comment by kennethshsu** [comment]
Oh, I don't haha... Hmm, how about:

```python
friedland_gl_self_insurer = cl.load_sample("friedland_gl_self_insurer")

tort = pd.DataFrame({
    "EffDate": ["2003-01-01", "2004-01-01", "2005-01-01"],
    "RateChange": [0.67, 0.75, 0.00],
})

olf = cl.ParallelogramOLF(
    rate_history=tort,
    change_col="RateChange",
    date_col="EffDate",
    policy_length=12,
    approximation_grain="M",
    vertical_line=True,
    change_factor=False # new parameter
).fit_transform(friedland_gl_self_insurer["Reported Claims"]).olf_
assert  np.all(olf.to_frame().values.flatten() == [0.67, 0.67, 0.67, 0.67, 0.67, 0.75, 1.  , 1.  , 1.  , 1.  , 1.  ])
```

I don't like the way `date_col` is tied to the end of the change though...

**Comment by henrydingliu** [comment]
>I don't like the way date_col is tied to the end of the change though...

can you elaborate? is it not determined by the X from fit_transform?

**Comment by kennethshsu** [comment]
> > I don't like the way date_col is tied to the end of the change though...
> 
> can you elaborate? is it not determined by the X from fit_transform?

Hmm just look at the example in the text vs the implementation of `cl.ParallelogramOLF()`. The intent of the current implementation is to bring rate level to current, if we use it to adjust loss/premium effect, the date is a bit weird. I would think something like below would be more intuitive. 

We would basically just need to generalize the parameters? I'd rather not deprecate anything and just make it more robust.

```python
friedland_gl_self_insurer = cl.load_sample("friedland_gl_self_insurer")

tort = pd.DataFrame({
    "EffDate": [("2000-01-01", "2003-12-31"), ("2004-01-01", "2004-12-31"), ("2005-01-01", "2005-12-31")],
    "RateChange": [0.67, 0.75, 0.00],
})

olf = cl.ParallelogramOLF(
    rate_history=tort,
    change_col="RateChange",
    date_col="EffDate",
    policy_length=12,
    approximation_grain="M",
    vertical_line=True,
    change_factor=False # new parameter
).fit_transform(friedland_gl_self_insurer["Reported Claims"]).olf_
assert np.all(olf.to_frame().values.flatten() == [0.67, 0.67, 0.67, 0.67, 0.67, 0.75, 1.  , 1.  , 1.  , 1.  , 1.  ])
```

**Comment by henrydingliu** [comment]
i don't use parallelogramolf a lot. so i don't know if it needs a refactor. so maybe we don't have to touch parallelgramolf directly? 

maybe this is the time to finally get trendconstant stood up? we can add a parameter like `cumulative`. if `cumulative=True`, then we return a parallelogramolf underneath the hood. if `cumulative=False`, then we return a `Trend` underneath the hood.

**Comment by priyam0k** [comment]
i'd like to pick this one up. pushed a rough prototype to see if the direction is right.

turns out the change is small. `parallelogram_olf` builds a cumulative rate level with
`np.cumprod(1 + changes)` and then returns `crl / level`. an on-level factor is just the
reciprocal of that level, so supporting cumulative input is a matter of skipping the
cumprod and inverting. everything downstream (rolling average, vertical_line, leap year
handling) stays as is. i added a `cumulative` flag per @henrydingliu's wording, default
False, fully backwards compatible.

with it, the tort example becomes:

```python
tort = pd.DataFrame({
    "EffDate": ["1998-01-01", "2003-01-01", "2004-01-01"],
    "Factor": [0.67, 0.75, 1.00],
})

olf = cl.ParallelogramOLF(
    rate_history=tort,
    change_col="Factor",
    date_col="EffDate",
    policy_length=12,
    approximation_grain="M",
    vertical_line=True,
    cumulative=True,
).fit_transform(friedland_gl_self_insurer["Reported Claims"]).olf_

# [0.67, 0.67, 0.67, 0.67, 0.67, 0.75, 1.0, 1.0, 1.0, 1.0, 1.0]

**Comment by henrydingliu** [comment]
good by me. 

@kennethshsu thoughts?

**Comment by kennethshsu** [comment]
Same, let's implement!
